modules/packages/cargo/parser.go

```diff
@@ -136,15 +136,22 @@ func parsePackage(r io.Reader) (*Package, error) {
 
 	dependencies := make([]*Dependency, 0, len(meta.Deps))
 	for _, dep := range meta.Deps {
+		name := dep.ExplicitNameInToml
+		pkg := &dep.Name
+		if len(dep.ExplicitNameInToml) == 0 {
+			name = dep.Name
+			pkg = nil
+		}
```

Comment:
```suggestion
		// https://doc.rust-lang.org/cargo/reference/registry-web-api.html#publish
		// It is a string of the new package name if the dependency is renamed, otherwise empty
		name := dep.ExplicitNameInToml
		pkg := &dep.Name
		if name == "" {
			name = dep.Name
			pkg = nil
		}
```